game-core/src/main/java/games/strategy/triplea/delegate/battle/MustFightBattle.java

```diff
@@ -1479,40 +1472,57 @@ public void execute(final ExecutionStack stack, final IDelegateBridge bridge) {
-      steps.add(
-          new FirstStrikeDefendersFire() {
-            private static final long serialVersionUID = 99992L;
+      steps.add(defensiveFirstStrike);
+      new IExecutable() {
+        private static final long serialVersionUID = 99992L;
 
-            @Override
-            public void execute(final ExecutionStack stack, final IDelegateBridge bridge) {
-              firstStrikeDefendersFire(firstStrikeOrder.getDefender().getReturnFire());
-            }
-          });
+        @Override
+        public void execute(final ExecutionStack stack, final IDelegateBridge bridge) {
+          new DefensiveFirstStrike(
+                  MustFightBattle.this,
+                  MustFightBattle.this,
+                  FirstStrikeStepOrder.DEFENDER_SNEAK_ATTACK,
+                  returnFireAgainstDefendingSubs)
+              .execute(stack, bridge);
+        }
+      };
     }
     if (firstStrikeOrder.getAttacker() != FirstStrikeStepOrder.NOT_APPLICABLE) {
-      steps.add(
-          new FirstStrikeAttackersFire() {
-            private static final long serialVersionUID = 99991L;
+      steps.add(offensiveFirstStrike);
+      new IExecutable() {
+        private static final long serialVersionUID = 99991L;
 
-            @Override
-            public void execute(final ExecutionStack stack, final IDelegateBridge bridge) {
-              firstStrikeAttackersFire(firstStrikeOrder.getAttacker().getReturnFire());
-            }
-          });
+        @Override
+        public void execute(final ExecutionStack stack, final IDelegateBridge bridge) {
+          new OffensiveFirstStrike(
+                  MustFightBattle.this,
+                  MustFightBattle.this,
+                  // the order doesn't matter at this point
+                  FirstStrikeStepOrder.OFFENDER_SNEAK_ATTACK,
+                  returnFireAgainstAttackingSubs)
+              .execute(stack, bridge);
+        }
+      };
     }
     if (firstStrikeOrder.getDefender()
             == FirstStrikeStepOrder.DEFENDER_SNEAK_ATTACK_WITH_OPPOSING_FIRST_STRIKE
         || firstStrikeOrder.getDefender()
             == FirstStrikeStepOrder.DEFENDER_NO_SNEAK_ATTACK_BUT_BEFORE_STANDARD_ATTACK) {
-      steps.add(
-          new FirstStrikeDefendersFire() {
-            private static final long serialVersionUID = 99992L;
+      steps.add(defensiveFirstStrike);
+      new IExecutable() {
+        private static final long serialVersionUID = 99992L;
 
-            @Override
-            public void execute(final ExecutionStack stack, final IDelegateBridge bridge) {
-              firstStrikeDefendersFire(firstStrikeOrder.getDefender().getReturnFire());
-            }
-          });
+        @Override
+        public void execute(final ExecutionStack stack, final IDelegateBridge bridge) {
+          new DefensiveFirstStrike(
+                  MustFightBattle.this,
+                  MustFightBattle.this,
+                  // The order doesn't matter at this point
```

Comment:
I'm not sure this comment makes sense, what is "this point" relative to? Secondly, the ordering of what exactly, the step order?

Reply:
This is talking about the next parameter.  I was trying to explain why I could pick a "random" `order`.  To construct the class, I need an `order` but at this point in the code, I can pick one of several.  I would prefer to use `firstStrikeOrder.getDefender()` but `firstStrikeOrder` doesn't exist in an old save and a null point exception would be thrown.

So, maybe I could change the comment to something like
```
firstStrikeOrder.getDefender() is the correct value but that isn't available in an old save so one of the possible FirstStrikeStepOrder.FirstStrikeResult has to be choosen
```